src/components/HydrantList.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import List, { ListItem, ListItemText } from 'material-ui/List';
+import Delete from '@material-ui/icons/Delete';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import Collapse from 'material-ui/transitions/Collapse';
+import Input, { InputLabel } from 'material-ui/Input';
+import { withStyles } from 'material-ui/styles';
+import ModeEdit from '@material-ui/icons/ModeEdit';
+import { FormControl, FormHelperText } from 'material-ui/Form';
+import TextField from 'material-ui/TextField';
+import _ from 'lodash';
+
+
+const styles = theme => ({
+  root: {
+    flexWrap: 'wrap'
+  },
+  inset: {
+    paddingLeft: 0
+  }
+});
+
+
+
+class HydrantList extends React.Component {
+
+  state = {
+    open: true,
+    showDetails:{}
+  }
+
+render() {
+
+  const {
+    classes,
+    editableTrail,
+    hydrants,
+    hydrantDeleted,
+    modifyHydrant
+  } = this.props;
+
+  const trailHydrants = hydrants.filter((h) => h.get('trail') === editableTrail.get('id')).valueSeq();
+
+  const toggleHighLight = (feature, state) => {
+    feature.set('highlighted', state)
+    feature.changed();
+  }
+
+  const updateCoords = (e, h, coordIndex) => {
+    const newCoords = _.clone(h.get('coords'));
+    newCoords.splice(coordIndex, 1, Number(e.target.value))
+    modifyHydrant(h.get('id'), { coords: newCoords })
+  };
+
+  const toggleEdit = (id) => {
+    const newDetails = {}
+    if (!this.state.showDetails[id]){
+      newDetails[id] = true
+    }
+    this.setState({ showDetails: newDetails })
+  }
+
+  const hydrantsList = trailHydrants.map((h, index)=> {
+    return (
```

Comment:
More readable and better to separate this into a function; `(h, index) => renderHydrant(h, index)`;

Good rule is probably to always have things that render separately as separate functions, if not entirely separate components. I've noted before most projects will just make a lot more different component files than we have here. If it's purely used internal to another component I'm fine leaving it in the class, but things that are mapped or calculated separately should be separate too. Render functions themselves should be as lightweight and straightforward as possible.

Reply:
Good Call. Check.